packages/camera/camera_avfoundation/lib/src/avfoundation_camera.dart

```diff
@@ -248,14 +248,23 @@ class AVFoundationCamera extends CameraPlatform {
 
   @override
   Future<void> startVideoRecording(int cameraId,
-      {Duration? maxVideoDuration}) async {
+      {Duration? maxVideoDuration,
+      Function(CameraImageData image)? streamCallback,
+      CameraImageStreamOptions? streamOptions}) async {
     await _channel.invokeMethod<void>(
       'startVideoRecording',
       <String, dynamic>{
         'cameraId': cameraId,
         'maxVideoDuration': maxVideoDuration?.inMilliseconds,
+        'enableStream': streamCallback != null,
       },
     );
+
+    if (streamCallback != null) {
+      _installStreamController();
+      _frameStreamController!.stream.listen(streamCallback);
```

Comment:
can you comment out why _frameStreamController must not be nil?

Reply:
Good point. I will actually change `_installStreamController` to return the controller, and then call listen on that. Will make the code easier to read, and remove the need for a `!`.